File: examples/workspaces/playwright-polyglot-example/java/src/main/java/Check.java
// Playwright, the Java way (com.microsoft.playwright). Open a real page, screenshot it,
// print TITLE|SUM.
import java.nio.file.Paths;

import com.microsoft.playwright.Browser;
import com.microsoft.playwright.Page;
import com.microsoft.playwright.Playwright;
import com.microsoft.playwright.options.WaitUntilState;

public class Check {
    public static void main(String[] args) {
        String url = System.getenv().getOrDefault("PAGE_URL", "https://www.java.com/en/");
        String shot = System.getenv().getOrDefault("SHOT_PATH", "java.png");

        try (Playwright pw = Playwright.create()) {
            Browser browser = pw.chromium().launch();
            Page page = browser.newPage(new Browser.NewPageOptions()
                    .setViewportSize(1280, 800));
            page.navigate(url, new Page.NavigateOptions()
                    .setWaitUntil(WaitUntilState.DOMCONTENTLOADED)
                    .setTimeout(60000));
            page.waitForTimeout(3000);
            Object sum = page.evaluate(
                    "() => { let s = 0; for (let i = 1; i <= 10; i++) s += i; return s; }");
            String title = page.title();
            page.screenshot(new Page.ScreenshotOptions().setPath(Paths.get(shot)));
            System.out.println(title + "|" + sum);
            browser.close();
        }
    }
}

File: examples/workspaces/playwright-polyglot-example/go/check.go
// Playwright, the Go way (community binding playwright-community/playwright-go).
// Open a real page, screenshot it, print TITLE|SUM. Unlike the four official bindings,
// playwright-go pins its own driver version, so it installs its own browser on first run
// rather than sharing the pre-baked one.
package main

import (
	"fmt"
	"os"

	// The project lives at github.com/playwright-community/playwright-go, but its
	// module path is still the original author's, so that's what we import (the
	// package itself is named "playwright").
	playwright "github.com/mxschmitt/playwright-go"
)

func main() {
	url := getenv("PAGE_URL", "https://go.dev/")
	shot := getenv("SHOT_PATH", "go.png")

	// Fetch the driver + chromium for this binding's pinned version (idempotent).
	if err := playwright.Install(&playwright.RunOptions{Browsers: []string{"chromium"}}); err != nil {
		panic(err)
	}

	pw, err := playwright.Run()
	if err != nil {
		panic(err)
	}
	browser, err := pw.Chromium.Launch()
	if err != nil {
		panic(err)
	}
	page, err := browser.NewPage(playwright.BrowserNewPageOptions{
		Viewport: &playwright.Size{Width: 1280, Height: 800},
	})
	if err != nil {
		panic(err)
	}
	if _, err = page.Goto(url, playwright.PageGotoOptions{
		WaitUntil: playwright.WaitUntilStateDomcontentloaded,
		Timeout:   playwright.Float(60000),
	}); err != nil {
		panic(err)
	}
	page.WaitForTimeout(3000)
	sum, err := page.Evaluate("() => { let s = 0; for (let i = 1; i <= 10; i++) s += i; return s; }")
	if err != nil {
		panic(err)
	}
	title, err := page.Title()
	if err != nil {
		panic(err)
	}
	if _, err = page.Screenshot(playwright.PageScreenshotOptions{
		Path: playwright.String(shot),
	}); err != nil {
		panic(err)
	}
	fmt.Printf("%s|%v\n", title, sum)

	_ = browser.Close()
	_ = pw.Stop()
}

func getenv(key, def string) string {
	if v := os.Getenv(key); v != "" {
		return v
	}
	return def
}
